opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx

```diff
@@ -198,6 +198,34 @@ export const removeEntityTypeAllFromFilterGroup = (inputFilters?: FilterGroup) =
   return inputFilters;
 };
 
+// fetch the entity type filters possible values of first and second levels
+// and remove Observable if the filters target only some sub observable types
+// exemple: Observable AND (Domain-Name) --> [Domain-Name]
+// exemple: Domain-Name OR Observable --> [Domain-Name, Observable]
+// exemple: Stix-Domain-Object AND (Malware OR (Country AND Location)) --> [Stix-Domain-Object, Malware]
+export const getEntityTypeTwoFirstLevelsFilterValues = (filters?: FilterGroup, observableTypes?: string[], domainObjectTypes?: string []) => {
+  if (!filters) {
+    return [];
+  }
+  let firstLevelValues = findFilterFromKey(filters.filters, 'entity_type', 'eq')?.values ?? [];
+
+  if (filters.mode === 'and') {
+    const subFilters = filters.filterGroups.map((fg) => fg.filters).flat();
```

Comment:
subFiltersSeparatedWithAnd = filters.filterGroups.filter((fg) => fg.mode === 'and').map((fg) => fg.filters).flat();